### PR TITLE
Role datums stat stufff

### PR DESCRIPTION
**Pull request by MadmanMartian**
Fixes up the stat_collector to work with factions and roles, rather than antags and roles.

Note to self: Update soods stuff once this is in and we're about to merge datums

Links are https://github.com/gbasood/vgstation-statistics-viewer/blob/master/app/models.py and https://github.com/gbasood/vgstation-statistics-viewer/blob/master/app/parsers/jsonparser.py

### CODE REVIEW

**Comment by gbasood** [comment]
So I've stared at this a bit more this time, and I think we should handle the gamemode-specific stat stuff differently now that we're adding datums.
Each antag group/gamemode/whatever the fuck should each be an object in an array in the JSON output, with the only exceptions being things that may happen without gamemodey things. Ayys, I think, can spawn without anything antag related showing up, but blobs probably will have a faction made for them on spawn, so ayys can stay how they are but blobs should go into a new format.

I haven't thought fully through how the database side of things should look yet, but I can work on a simple ERD later

**Comment by gbasood** [comment]
Also, not sure if it exists yet, but for factions that  are non-unique, like two groups of competing factions of the same type, it might be necessary to have a unique identifier for their object so we know what antag goes with what group.
Or store antags in an array inside the faction group instead, so there's no confusing what they belong to. Which might be how it already works, don't know, brainstorming before heading out to work so I'm not gonna re-read the JSON to check that

**Comment by DamianX** [comment]
Should probably reopen this targeting Bleeding-Edge

File: code/datums/statistics/stat_blob.dm [hunk 1]
```diff
@@ -18,8 +18,10 @@
 
 /datum/stat_blob/cult/doPostRoundChecks()
 	for(var/datum/mind/M in ticker.minds)
-		if(M.active && istype(M.current, /mob/living/carbon) && M.special_role == "Cultist")
-			surviving_cultists++
+		if(M.active && istype(M.current, /mob/living/carbon))
+			var/mob/living/carbon/C = M.current
+			if(islegacycultist(C))
```
Comment (gbasood) [inline code/datums/statistics/stat_blob.dm:23]: ~~Deity's PR that exports stats for dynamic mode info is going to remove this cultist check, not sure if we want to keep your rewritten one or place it elsewhere~~ future PR fixer: don't keep this

**Review by gbasood — CHANGES_REQUESTED** [review]
Alright, specific items that need changing:
 - Cult-specific stats should be logged under their faction's stat datum instead of directly on the stat_collection, so we can track what faction did what. This does not include blood spread on station, btw
 - Same as above but for every faction we care about.

Here's a faction checklist off the top of my head (including things that may not yet have factions but should, eventually, in the JSON output):
 - [ ] Nukeops
 - [ ] All types of cult currently in rotation
 - [ ] Malf (I think malf borgs will be members of this faction? but I'm not in the loop)
 - [ ] Ayys
 - [ ] Blob
 - [ ] Revs
 - [ ] Wizards (tracking spells bought ideally)
 - I'm probably forgetting things, things like ling/traitor don't necessarily need on-faction stats aside from what you've already written, but being able to track things like what powers a ling has unlocked at round end would be neato

**Pull request closed without merging**